src/sagas/wallet.js

```diff
@@ -350,13 +350,22 @@ export function* loadTokens() {
   return registeredUids;
 }
 
+/**
+ * It handles the request to fetch tokens metadata and open a thread to process it.
+ * @param {{ tokens: string[] }}
+ */
+export function* fetchTokensMetadataRequest(action) {
+  yield spawn(fetchTokensMetadata, action.tokens);
+}
+
 /**
  * The wallet needs each token metadata to show information correctly
  * So we fetch the tokens metadata and store on redux
  */
 export function* fetchTokensMetadata(tokens) {
-  // No tokens to load
+  // No tokens to load, set metadata as loaded
   if (!tokens.length) {
+    yield put(tokenMetadataUpdated({}));
```

Comment:
Why are you passing an empty object?

Reply:
We should just return

Reply:
This use of `tokenMetadataUpdated` results the state `metadataLoaded` to be `true`. We need this state in `NewNanoContractTransactionRequest` to [evaluate some flags](https://github.com/HathorNetwork/hathor-wallet-mobile/pull/495/files#diff-f4c181ecb3766d7252d1d68de29c6e0431178ae696108807b41011b64feef35fR163-R169).

Reply:
Passing an empty object has no effect for the `tokenMetadata` state as we can [see here](https://github.com/HathorNetwork/hathor-wallet-mobile/blob/12e4539d85afdc2d61a2e99813a516414e5cf9cc/src/reducers/reducer.js#L913).

Reply:
Never mind. I refactored it and the `metadataLoaded` is not being used anymore. I implement now a flag `knownTokens.isLoading` to evaluate those flags and control component rendering.